dulwich/repo.py

```diff
@@ -1234,6 +1246,14 @@ def _determine_file_mode(self):
 
         return mode_differs and st2_has_exec
 
+    def _determine_symlinks(self):
+        """Probe the filesystem to determine whether symlinks can be created.
+
+        Returns: True if symlinks can be created, False otherwise.
+        """
+        # TODO(jelmer): Actually probe disk / look at filesystem
```

Comment:
```
with TemporaryDirectory() as tmpdir:
    source = Path(tmpdir) / 'file.txt'
    source.touch()

    link = Path(tmpdir) / 'link.txt'

    try:
        link.symlink_to(source)
        return True
    except OSError:
        return False
```

Reply:
Good point.

However, on windows, it depends on setted privileges:
> On newer versions of Windows 10, unprivileged accounts can create symlinks if Developer Mode is enabled. When Developer Mode is not available/enabled, the SeCreateSymbolicLinkPrivilege privilege is required, or the process must be run as an administrator.

I'm not sure it is possible to probe without trying to create a real file.

But deactivating symlink on windows unless specifically required is already more that enough. Thank you!